client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/Allocation.tsx

```diff
@@ -41,14 +43,18 @@ export const Allocation = ({
     })
   );
 
+  const { data: sortByVvmStatus } = usePreference(
+    PreferenceKey.SortByVvmStatusThenExpiry
+  );
+
   useEffect(() => {
     initialise(
       itemData,
-      AllocationStrategy.FEFO,
+      sortByVvmStatus ? AllocationStrategy.VVMStatus : AllocationStrategy.FEFO,
       allowPlaceholder,
       scannedBatch
     );
-  }, []);
+  }, [sortByVvmStatus]);
```

Comment:
Not sure if this is ok to do? It seems to fix the issue where sometimes the sort order wouldn't apply.
I think it's probably not ideal, so do we need to pass it in to this component rather than checking the pref here? I think it's convient here but maybe unintended consequences?

Reply:
#7763 added support for multi pref query, at which point, so have put that in the parent component. Then the pref query only runs once, when the modal first opens, rather than for each item as you click next...

Maybe this is still wise though, in case prefs somehow take a long time to resolve? Would just initialise the modal twice 🤷‍♀️